Source/CardStackController.m

```diff
@@ -522,60 +511,86 @@ - (void)updateCardTitleBarBackgroundColors {
 }
 
 - (void)updateCardLocations {
-    if (self.isSeachViewControllerHidden) {
-        CGRect frame = self.searchViewController.view.frame;
-        frame.origin.y = -self.searchViewController.view.bounds.size.height;
-        self.searchViewController.view.frame = frame;
+    [self updateCardLocationsAnimated:NO];
+}
 
+- (void)updateCardLocationsAnimated:(BOOL)animated {
+    if (animated) {
+        [self updateCardLocationsAnimatedWithVerticalVelocity:0.0f];
+    } else {
+        for (NSUInteger i = 0; i < self.cards.count; i++) {
+            CardView *card = [self.cards objectAtIndex:i];
+            card.frame = [self frameForCardAtIndex:i];
+        }
+    }
+}
+
+- (void)updateCardLocationsAnimatedWithVerticalVelocity:(CGFloat)verticalVelocity {
+    for (NSUInteger i = 0; i < self.cards.count; i++) {
+        CardView *card = [self.cards objectAtIndex:i];
+        POPSpringAnimation *springAnimation = [POPSpringAnimation animationWithPropertyNamed:kPOPViewFrame];
+        CGRect frame = [self frameForCardAtIndex:i];
+        springAnimation.toValue = [NSValue valueWithCGRect:frame];
+        springAnimation.springBounciness = 8;
+        if (verticalVelocity > 0.0f && i <= self.currentCardIndex) {
```

Comment:
Would be better if you named this `if` instead of adding comments bellow

```objc
BOOL thisShouldBeDoneBecause = (....);
if (thisShouldBeDoneBecause) {
    // ...
} else {
    // ...
}
```